Orange/evaluation/testing.py

```diff
@@ -351,89 +381,17 @@ def __init__(self, data, learners, store_data=False, store_models=False,
         super().__init__(data, len(learners), store_data=store_data,
                          store_models=store_models, preprocessor=preprocessor,
                          callback=callback)
-        domain = data.domain
-        X = data.X.copy()
-        Y = data._Y.copy()
-        metas = data.metas.copy()
-
-        teX, trX = X[:1], X[1:]
-        teY, trY = Y[:1], Y[1:]
-        te_metas, tr_metas = metas[:1], metas[1:]
-        if data.has_weights():
-            W = data.W.copy()
-            teW, trW = W[:1], W[1:]
-        else:
-            W = teW = trW = None
+        self.learners = learners
 
-        self.row_indices = np.arange(len(data))
-        if self.store_models:
-            self.models = []
-        self.actual = Y.flatten()
-        nmethods = len(learners)
-        n_callbacks = nmethods * len(data)
-        for test_idx in self.row_indices:
-            X[[0, test_idx]] = X[[test_idx, 0]]
-            Y[[0, test_idx]] = Y[[test_idx, 0]]
-            metas[[0, test_idx]] = metas[[test_idx, 0]]
-            if W:
-                W[[0, test_idx]] = W[[test_idx, 0]]
-            test_data = Table.from_numpy(domain, teX, teY, te_metas, teW)
-            train_data = Table.from_numpy(domain, trX, trY, tr_metas, trW)
-            if self.preprocessor is not None:
-                train_data = self.preprocessor(train_data)
-            if self.store_models:
-                fold_models = [None] * nmethods
-                self.models.append(fold_models)
-            for i, learner in enumerate(learners):
-                model = self.train_if_succ(i, learner, train_data)
-                self.call_callback((test_idx * nmethods + i) / n_callbacks)
-                if not model:
-                    continue
-                if self.store_models:
-                    fold_models[i] = model
-                if data.domain.has_discrete_class:
-                    values, probs = model(test_data, model.ValueProbs)
-                    self.predicted[i][test_idx] = values
-                    self.probabilities[i][test_idx, :] = probs
-                elif data.domain.has_continuous_class:
-                    values = model(test_data, model.Value)
-                    self.predicted[i][test_idx] = values
-        self.call_callback(1)
+        self.indices = skl_cross_validation.LeaveOneOut(len(data))
 
+        if self.preprocessor is not None:
+            data = self.preprocessor(data)
```

Comment:
It can do anything, in principle. If it does anything based on values computed from the data, it is wrong to use the entire data set. If it, for instance, subtracts the mean, this mean has to be computed from training data only. It must not see, use or touch the test data in any way.